scripts/host-check

```diff
@@ -1126,31 +1139,32 @@ def check_iommu() -> CheckFuncReturn:
         return (
             CheckState.ERROR,
             f"The cmd {cmd!r} failed - unable to\n"
-            "determine the network devices on the host. IOMMU is enabled.",
-        )
-    net_iommu_devices = network_devices & set(iommu_devices)
-    if not net_iommu_devices:
-        return (
-            warning_state,
-            "IOMMU enabled for vfio-pci, but no network PCI devices found.",
-        )
-    net_iommu_devs_list = []
-    for device in sorted(net_iommu_devices):
-        net_iommu_devs_list.append(
-            net_devices_dict[device] + " (" + device + ")"
+            "determine the network devices on the host.",
         )
+
+    # If in IOMMU mode then check if the network devices are in the IOMMU group
+    if iommu_devices:
```

Comment:
`iommu_devices` may not be defined I think? (if we fall into the `except` on L1119)

Reply:
(if so there should be a test to cover this)

Reply:
Changed the except so that iommu_devices = [] if the directory search fails.

Reply:
Resolving this - I'm not sure what the correct behaviour should be here if there are no IOMMU devices, but moving this to a different thread.